src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -662,6 +664,28 @@ namespace winrt::TerminalApp::implementation
         return _settings.GlobalSettings().CurrentTheme();
     }
 
+    bool AppLogic::IsolatedMode()
+    {
+        if (!_loadedInitialSettings)
+        {
+            // Load settings if we haven't already
+            ReloadSettings();
+        }
+        return _settings.GlobalSettings().IsolatedMode();
+    }
+    bool AppLogic::RequestsTrayIcon()
+    {
+        if (!_loadedInitialSettings)
+        {
+            // Load settings if we haven't already
+            ReloadSettings();
+        }
+
+        return _settings.GlobalSettings().AlwaysShowNotificationIcon() ||
+               _settings.GlobalSettings().MinimizeToNotificationArea();

```

Comment:
Could you introduce a local variable for `_settings.GlobalSettings()`? It's not terrible if you don't of course, since the impact is absolutely minimal in this one-off case, but I feel like that avoiding repeated WinRT calls across our entire code base consistently could have a noticeable long-term impact.